src/java_binding/src/lib.rs

```diff
@@ -425,6 +454,66 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetStringValu
     })
 }
 
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetTimestampValue<'a>(
+    env: EnvParam<'a>,
+    pointer: Pointer<'a, JavaBindingRow>,
+    idx: jint,
+) -> JObject<'a> {
+    // since JMethodID is always validate until the belonging class is unload.
+    static INIT_METHOD: OnceCell<JMethodID> = OnceCell::new();
+    execute_and_catch(env, move || {
+        let millis = pointer
+            .as_ref()
+            .get_datetime(idx as usize)
+            .0
+            .timestamp_millis();
+        let ts_class_ref = pointer
+            .as_ref()
+            .class_cache
+            .timestamp_class
+            .get_or_try_init(|| {
+                let cls = env.find_class("java/sql/Timestamp")?;
+                env.new_global_ref(cls)
+            })?;
+        let ts_class = JClass::from(ts_class_ref.as_obj());
+        let constructor =
+            INIT_METHOD.get_or_try_init(|| env.get_method_id(ts_class, "<init>", "(J)V"))?;
+        let date_obj = env.new_object_unchecked(ts_class, *constructor, &[millis.into()])?;
+
+        Ok(date_obj)
+    })
+}
+
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_rowGetDecimalValue<'a>(
+    env: EnvParam<'a>,
+    pointer: Pointer<'a, JavaBindingRow>,
+    idx: jint,
+) -> JObject<'a> {
+    static INIT_METHOD: OnceCell<JMethodID> = OnceCell::new();
```

Comment:
Any reason to use a static variable to store the method instead of storing it in the row cache?

Reply:
The JMethodID is globally available until the refered class is unload. But if the refered class is unload, we will result to an error then return when calling `find_class` and never have chance to access the INIT_METHOD. So it's safe here to cache it with static lifetime. I will write more detailed comment.

Reply:
The code itself is safe. It just looks strange that we find class for every iterator, but we keep a global static variable to store the method id and initialize it for only once. 

It is cleaner to store the class and the method id together in the row cache and initialize and gets dropped together.

Reply:
I have written comment to explain it and it's not hard to understand, I think.

Reply:
This is not a discussion about safety. What I mean is, the method id is associated with the class because it is initialized from the class. Therefore, to keep the code clean, the method id is better to be stored together with the class itself. 

We can either store the method id with the class object together in the `ClassMethodCache`, and initialize both of them in every iteration creation, or we store both the method id and the class object as static variable and globally get initialized for only once.

Btw, any reason not to store the class object also as a static variable? It is already global reference, and IIUC it's safe to store as static variable, and then we don't need the extra `ClassMethodCache`.

Reply:
The JClass and JMethodID are distinct things, and there is no need to treat them as same.

JClass is extended from JObject, it's a reference type, that means it owned a reference to a object managed by JVM. If we do not properly destroy it, it can result in a memory leak. ( JVM GC can not free it )

JMethodID is a regular pointer type, there's no corresponding entity that managed by JVM.

The reason why I create a struct called `ClassMethodCache` is to bind the JClasses' lifetime to the iterator so they can be destroyed together with the iterator, then the JVM GC could free those objects.

Since JMethodID is not managed by JVM GC, there's no need to bind it to some scope.